libs/@local/hash-isomorphic-utils/src/save.ts

```diff
@@ -141,12 +136,11 @@ const calculateSaveActions = async (
           throw new Error("Cannot find parent entity");
         }
 
-        const [assumedEntityTypeId, newTypeActions] =
-          await getEntityTypeForComponent(blockEntity.componentId ?? "");
+        const assumedEntityTypeId = getEntityTypeForComponent(
+          blockEntity.componentId ?? "",
```

Comment:
This is still not great as the `componentId` is optional, not sure if we can do anything about this without investing more time cleaning up the entity store